src/pages/Deposit.js

```diff
@@ -43,36 +65,58 @@ export default function Deposit() {
     console.log(response);
   }
 
-  async function reloadTotalDepositAmount() {
-    const totalBN = await contract.totalDepositedAmount();
-    const total = ethers.utils.formatEther(totalBN);
-    setTotalDepositedAmount(total);
-  }
+  const handleAddressChange = (e) => {
+    const input = e.target.value;
+    setContractAddress(input && input.length ? input : null);
```

Comment:
@akinwol Contract is loaded via url and state from previous page. However, I cannot really let contract loaded when address is entered in the form. I'm not sure how `useProjectContract`, `useMemo` are supposed to work. Is it something you can help, please?

Reply:
@charlie-kim thank you so much for your work!!!!
seems like setContractAddress context value actually takes a dictionary looks like this
`contractAddress: { address: "", projectName: "", symbol: "" },`

so i guess it should be something like
```suggestion
    setContractAddress({...contracatAddress, address: input && input.length ? input : null});
```

Reply:
we might rename it to be more like, `contractMetadata` since it holds more info than just address https://github.com/PlaygroundDapp/playground/blob/14ba947387b0f98a3c68cabcc37a39759d8f65b9/src/context/Web3Context.js#L11

Reply:
updated